Ensure at grammar level that module application is always to a qualid. (#15355)
coq/coq: doc/changelog/07-vernac-commands-and-options/15355-mod-apply-ident.rst

- **Changed:**

>   the syntax of :cmd:`Module` only admits parentheses around module arguments. For instance ``Module M := (F X).` is now a parsing error

Reviewer: ```suggestion
  the syntax of :cmd:`Module` only admits parentheses around module arguments. For instance `Module M := (F X).` is now a parsing error
```

>   the syntax of :cmd:`Module` only admits parentheses around module arguments. For instance ``Module M := (F X).`` is now a parsing error

Reviewer 2: ```suggestion
  :cmd:`Module` now only allows parentheses around module arguments. For instance, ``Module M := (F X).`` is now a parsing error
```
